src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -680,5 +710,307 @@ private async Task<string> GenerateToken(ClaimsPrincipal principal, DateTime? ex
+                // General OIDC claims
+                if (claim.Type.Equals("nonce"))
+                {
+                    userAuthenticationModel.Nonce = claim.Value;
+                    continue;
+                }
+
+                // Altinn Specific claims
+                if (claim.Type.Equals(AltinnCoreClaimTypes.UserId))
+                {
+                    userAuthenticationModel.UserID = Convert.ToInt32(claim.Value);
+                    continue;
+                }
+
+                if (claim.Type.Equals(AltinnCoreClaimTypes.PartyID))
+                {
+                    userAuthenticationModel.PartyID = Convert.ToInt32(claim.Value);
+                    continue;
+                }
+
+                if (claim.Type.Equals(AltinnCoreClaimTypes.AuthenticateMethod))
+                {
+                    userAuthenticationModel.AuthenticationMethod = (Enum.AuthenticationMethod)System.Enum.Parse(typeof(Enum.AuthenticationMethod), claim.Value);
+                    continue;
+                }
+
+                if (claim.Type.Equals(AltinnCoreClaimTypes.AuthenticationLevel))
+                {
+                    userAuthenticationModel.AuthenticationLevel = (Enum.SecurityLevel)System.Enum.Parse(typeof(Enum.SecurityLevel), claim.Value);
+                    continue;
+                }
+
+                // ID-porten specific claims
+                if (claim.Type.Equals("pid"))
+                {
+                    userAuthenticationModel.SSN = claim.Value;
+                    continue;
+                }
+
+                if (claim.Type.Equals("amr"))
+                {
+                    userAuthenticationModel.AuthenticationMethod = GetAuthenticationMethod(claim.Value);
+                    continue;
+                }
+
+                if (claim.Type.Equals("acr"))
+                {
+                    userAuthenticationModel.AuthenticationLevel = GetAuthenticationLevel(claim.Value);
+                    continue;
+                }
+
+                // General claims handling
+                if (provider.ProviderClaims != null && provider.ProviderClaims.Contains(claim.Type))
+                {
+                    userAuthenticationModel.ProviderClaims.Add(claim.Type, claim.Value);
+                }
+            }
+
+            return userAuthenticationModel;
+        }
+
+        /// <summary>
+        /// Converts IDporten acr claim “Authentication Context Class Reference” - The security level of assurance for the
+        /// authentication. Possible values are Level3 (i.e. MinID was used) or Level4 (other eIDs).
+        /// The level must be validated by the client.
+        /// </summary>
+        private static SecurityLevel GetAuthenticationLevel(string acr)
+        {
+            switch (acr)
+            {
+                case "Level3":
+                    return Enum.SecurityLevel.Sensitive;
+                case "Level4":
+                    return Enum.SecurityLevel.Sensitive;
```

Comment:
Assuming it is on purpose that both levels give the same security level I think it should be commented.

Reply:
Fixed